### ISSUE TITLE
4070ti 10个小时了还没预训练完正常吗？

### ISSUE THREAD

**Issue by MoMingQimio** [jingyaogong/minimind#772]
Model Params: 63.91M
Trainable Params: 63.912M
Generating train split: 1270238 examples [00:01, 786507.77 examples/s]
Epoch:[1/2](100/39695), loss: 7.5180, logits_loss: 7.5180, aux_loss: 0.0000, lr: 0.00050000, epoch_time: 4767.0min
Epoch:[1/2](200/39695), loss: 7.0277, logits_loss: 7.0277, aux_loss: 0.0000, lr: 0.00049999, epoch_time: 4750.0min
Epoch:[1/2](300/39695), loss: 6.6907, logits_loss: 6.6907, aux_loss: 0.0000, lr: 0.00049998, epoch_time: 4738.0min
Epoch:[1/2](400/39695), loss: 6.5465, logits_loss: 6.5465, aux_loss: 0.0000, lr: 0.00049997, epoch_time: 4725.0min
Epoch:[1/2](500/39695), loss: 6.2271, logits_loss: 6.2271, aux_loss: 0.0000, lr: 0.00049996, epoch_time: 4712.0min
Epoch:[1/2](600/39695), loss: 6.1334, logits_loss: 6.1334, aux_loss: 0.0000, lr: 0.00049994, epoch_time: 4699.0min
Epoch:[1/2](700/39695), loss: 5.9213, logits_loss: 5.9213, aux_loss: 0.0000, lr: 0.00049991, epoch_time: 4686.0min
Epoch:[1/2](800/39695), loss: 5.6375, logits_loss: 5.6375, aux_loss: 0.0000, lr: 0.00049989, epoch_time: 4674.0min
Epoch:[1/2](900/39695), loss: 5.4089, logits_loss: 5.4089, aux_loss: 0.0000, lr: 0.00049986, epoch_time: 4662.0min
Epoch:[1/2](1000/39695), loss: 5.2721, logits_loss: 5.2721, aux_loss: 0.0000, lr: 0.00049982, epoch_time: 4650.0min
Epoch:[1/2](1100/39695), loss: 4.9170, logits_loss: 4.9170, aux_loss: 0.0000, lr: 0.00049979, epoch_time: 4639.0min
Epoch:[1/2](1200/39695), loss: 4.6911, logits_loss: 4.6911, aux_loss: 0.0000, lr: 0.00049975, epoch_time: 4626.0min
Epoch:[1/2](1300/39695), loss: 4.5174, logits_loss: 4.5174, aux_loss: 0.0000, lr: 0.00049970, epoch_time: 4614.0min
Epoch:[1/2](1400/39695), loss: 4.6796, logits_loss: 4.6796, aux_loss: 0.0000, lr: 0.00049965, epoch_time: 4602.0min
Epoch:[1/2](1500/39695), loss: 4.0104, logits_loss: 4.0104, aux_loss: 0.0000, lr: 0.00049960, epoch_time: 4590.0min
Epoch:[1/2](1600/39695), loss: 4.2626, logits_loss: 4.2626, aux_loss: 0.0000, lr: 0.00049955, epoch_time: 4578.0min
Epoch:[1/2](1700/39695), loss: 3.9891, logits_loss: 3.9891, aux_loss: 0.0000, lr: 0.00049949, epoch_time: 4566.0min
Epoch:[1/2](1800/39695), loss: 3.9715, logits_loss: 3.9715, aux_loss: 0.0000, lr: 0.00049943, epoch_time: 4554.0min
Epoch:[1/2](1900/39695), loss: 3.8395, logits_loss: 3.8395, aux_loss: 0.0000, lr: 0.00049936, epoch_time: 4542.0min
Epoch:[1/2](2000/39695), loss: 3.7730, logits_loss: 3.7730, aux_loss: 0.0000, lr: 0.00049930, epoch_time: 4530.0min
Epoch:[1/2](2100/39695), loss: 3.8209, logits_loss: 3.8209, aux_loss: 0.0000, lr: 0.00049922, epoch_time: 4518.0min
Epoch:[1/2](2200/39695), loss: 3.7864, logits_loss: 3.7864, aux_loss: 0.0000, lr: 0.00049915, epoch_time: 4506.0min
Epoch:[1/2](2300/39695), loss: 3.7850, logits_loss: 3.7850, aux_loss: 0.0000, lr: 0.00049907, epoch_time: 4493.0min
Epoch:[1/2](2400/39695), loss: 3.3230, logits_loss: 3.3230, aux_loss: 0.0000, lr: 0.00049899, epoch_time: 4481.0min
Epoch:[1/2](2500/39695), loss: 3.7803, logits_loss: 3.7803, aux_loss: 0.0000, lr: 0.00049890, epoch_time: 4469.0min
Epoch:[1/2](2600/39695), loss: 3.7029, logits_loss: 3.7029, aux_loss: 0.0000, lr: 0.00049881, epoch_time: 4456.0min
Epoch:[1/2](2700/39695), loss: 3.4051, logits_loss: 3.4051, aux_loss: 0.0000, lr: 0.00049872, epoch_time: 4444.0min
Epoch:[1/2](2800/39695), loss: 3.2409, logits_loss: 3.2409, aux_loss: 0.0000, lr: 0.00049862, epoch_time: 4432.0min
Epoch:[1/2](2900/39695), loss: 3.5669, logits_loss: 3.5669, aux_loss: 0.0000, lr: 0.00049852, epoch_time: 4419.0min
Epoch:[1/2](3000/39695), loss: 3.3262, logits_loss: 3.3262, aux_loss: 0.0000, lr: 0.00049842, epoch_time: 4407.0min
Epoch:[1/2](3100/39695), loss: 2.9439, logits_loss: 2.9439, aux_loss: 0.0000, lr: 0.00049831, epoch_time: 4395.0min
Epoch:[1/2](3200/39695), loss: 3.1683, logits_loss: 3.1683, aux_loss: 0.0000, lr: 0.00049820, epoch_time: 4383.0min
Epoch:[1/2](3300/39695), loss: 3.0857, logits_loss: 3.0857, aux_loss: 0.0000, lr: 0.00049808, epoch_time: 4371.0min
Epoch:[1/2](3400/39695), loss: 3.2163, logits_loss: 3.2163, aux_loss: 0.0000, lr: 0.00049797, epoch_time: 4359.0min
Epoch:[1/2](3500/39695), loss: 3.1139, logits_loss: 3.1139, aux_loss: 0.0000, lr: 0.00049785, epoch_time: 4347.0min
Epoch:[1/2](3600/39695), loss: 3.0154, logits_loss: 3.0154, aux_loss: 0.0000, lr: 0.00049772, epoch_time: 4335.0min
Epoch:[1/2](3700/39695), loss: 3.0870, logits_loss: 3.0870, aux_loss: 0.0000, lr: 0.00049759, epoch_time: 4323.0min
Epoch:[1/2](3800/39695), loss: 2.8814, logits_loss: 2.8814, aux_loss: 0.0000, lr: 0.00049746, epoch_time: 4311.0min
Epoch:[1/2](3900/39695), loss: 3.0543, logits_loss: 3.0543, aux_loss: 0.0000, lr: 0.00049733, epoch_time: 4299.0min
Epoch:[1/2](4000/39695), loss: 2.7869, logits_loss: 2.7869, aux_loss: 0.0000, lr: 0.00049719, epoch_time: 4287.0min
Epoch:[1/2](4100/39695), loss: 2.8419, logits_loss: 2.8419, aux_loss: 0.0000, lr: 0.00049705, epoch_time: 4276.0min
Epoch:[1/2](4200/39695), loss: 2.7353, logits_loss: 2.7353, aux_loss: 0.0000, lr: 0.00049690, epoch_time: 4264.0min
Epoch:[1/2](4300/39695), loss: 2.9718, logits_loss: 2.9718, aux_loss: 0.0000, lr: 0.00049675, epoch_time: 4252.0min
Epoch:[1/2](4400/39695), loss: 2.8866, logits_loss: 2.8866, aux_loss: 0.0000, lr: 0.00049660, epoch_time: 4240.0min
Epoch:[1/2](4500/39695), loss: 3.0210, logits_loss: 3.0210, aux_loss: 0.0000, lr: 0.00049644, epoch_time: 4228.0min
Epoch:[1/2](4600/39695), loss: 3.0452, logits_loss: 3.0452, aux_loss: 0.0000, lr: 0.00049628, epoch_time: 4216.0min
Epoch:[1/2](4700/39695), loss: 2.7683, logits_loss: 2.7683, aux_loss: 0.0000, lr: 0.00049612, epoch_time: 4204.0min
Epoch:[1/2](4800/39695), loss: 2.9192, logits_loss: 2.9192, aux_loss: 0.0000, lr: 0.00049595, epoch_time: 4192.0min
Epoch:[1/2](4900/39695), loss: 3.0033, logits_loss: 3.0033, aux_loss: 0.0000, lr: 0.00049578, epoch_time: 4180.0min
Epoch:[1/2](5000/39695), loss: 2.7949, logits_loss: 2.7949, aux_loss: 0.0000, lr: 0.00049561, epoch_time: 4168.0min
Epoch:[1/2](5100/39695), loss: 2.7715, logits_loss: 2.7715, aux_loss: 0.0000, lr: 0.00049543, epoch_time: 4156.0min
Epoch:[1/2](5200/39695), loss: 2.6288, logits_loss: 2.6288, aux_loss: 0.0000, lr: 0.00049525, epoch_time: 4144.0min
Epoch:[1/2](5300/39695), loss: 2.6534, logits_loss: 2.6534, aux_loss: 0.0000, lr: 0.00049507, epoch_time: 4132.0min
Epoch:[1/2](5400/39695), loss: 2.6975, logits_loss: 2.6975, aux_loss: 0.0000, lr: 0.00049488, epoch_time: 4120.0min
Epoch:[1/2](5500/39695), loss: 3.0238, logits_loss: 3.0238, aux_loss: 0.0000, lr: 0.00049469, epoch_time: 4108.0min
Epoch:[1/2](5600/39695), loss: 2.7863, logits_loss: 2.7863, aux_loss: 0.0000, lr: 0.00049450, epoch_time: 4096.0min
Epoch:[1/2](5700/39695), loss: 2.6489, logits_loss: 2.6489, aux_loss: 0.0000, lr: 0.00049430, epoch_time: 4084.0min
Epoch:[1/2](5800/39695), loss: 2.6683, logits_loss: 2.6683, aux_loss: 0.0000, lr: 0.00049410, epoch_time: 4072.0min
Epoch:[1/2](5900/39695), loss: 2.8105, logits_loss: 2.8105, aux_loss: 0.0000, lr: 0.00049390, epoch_time: 4060.0min
Epoch:[1/2](6000/39695), loss: 2.7185, logits_loss: 2.7185, aux_loss: 0.0000, lr: 0.00049369, epoch_time: 4048.0min
Epoch:[1/2](6100/39695), loss: 2.5780, logits_loss: 2.5780, aux_loss: 0.0000, lr: 0.00049348, epoch_time: 4036.0min
Epoch:[1/2](6200/39695), loss: 2.7120, logits_loss: 2.7120, aux_loss: 0.0000, lr: 0.00049326, epoch_time: 4024.0min
Epoch:[1/2](6300/39695), loss: 2.5792, logits_loss: 2.5792, aux_loss: 0.0000, lr: 0.00049304, epoch_time: 4012.0min
Epoch:[1/2](6400/39695), loss: 2.5602, logits_loss: 2.5602, aux_loss: 0.0000, lr: 0.00049282, epoch_time: 4000.0min

一个epoch要4000分钟

**Comment by jingyaogong** [comment]
不正常，这个速度不可能在cuda上。自行检查
https://github.com/jingyaogong/minimind#%E2%85%B1-%EF%B8%8F-%E6%A8%A1%E5%9E%8B%E8%AE%AD%E7%BB%83
